### ISSUE TITLE
Not able to navigate using tab key when using nested Navigators

### ISSUE THREAD

**Issue by eduardoGo** [flutter/flutter#109624]
I'm not able to navigate using tab key when using nested Navigators, see the minimal reproducible code.

## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. Use tab key to navigate between widgets

**Expected results:** You should be able to navigate in the entire widget tree

**Actual results:** You can't navigate to the app bar button, neither the widgets of the outer navigator.

<details>
<summary>Minimal reproducible code</summary>

```dart
void main()  => runApp(MyApp());

class MyApp extends StatefulWidget {
  const MyApp({
    super.key,
  });

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Navigator(
        pages: [
          MaterialPage(
            child: Scaffold(
              appBar: AppBar(
                leading: IconButton(
                  icon: const Icon(Icons.menu),
                  onPressed: () {},
                ),
              ),
              body: Row(
                children: [
                  TextButton(
                    onPressed: () {},
                    child: const Text('Menu button 1'),
                  ),
                  Expanded(
                    child: Navigator(
                      pages: [
                        MaterialPage(
                          child: Column(
                            children: [
                              TextButton(
                                onPressed: () {},
                                child: const Text('Body button 1'),
                              ),
                              TextButton(
                                onPressed: () {},
                                child: const Text('Body button 2'),
                              ),
                            ],
                          ),
                        ),
                      ],
                    ),
                  ),
                ],
              ),
            ),
          ),
        ],
      ),
    );
  }
}
```

</details>

**Comment by jacobsimionato** [comment]
Updating to P2 because this is about to block a Google customer launch in September see internal issue b/241143126 for more context.

**Comment by jacobsimionato** [comment]
goderbauer@, could you please route this to someone who can look into it?

**Comment by goderbauer** [comment]
@gspencergoog This looks like a focus traversal problem. Could you take a look at this?

**Comment by gspencergoog** [comment]
I think this is because of the nested `Navigator` widgets. A `Navigator` adds a `FocusScope` around its contents, and that is what is restricting the traversal, so the inner navigator is preventing traversal outside of itself.

**Comment by jacobsimionato** [comment]
That makes sense! This behavior seems to be causing problems when the inner navigator does not occupy the entire screen. Do you have any ideas for workarounds or framework-level fixes?

**Comment by gspencergoog** [comment]
The first idea would be not to use nested navigators, if possible. If not, I can think of some possible framework level changes that might help, but are quite risky:

1) Simply remove the `FocusScope` from `Navigator`.  This probably would break a lot of tab navigation in a lot of apps, and probably wouldn't be acceptable.
2) Replace the `FocusScope` in the Navigator with a `FocusTraversalGroup`, and move the `FocusScope` outside of the Navigator in `WidgetsApp`, so that there's still a scope, but only one, not any nested ones.  The `FocusTraversalGroup` would help keep the tab order sane, but still allow traversal outside of the group.

**Comment by gspencergoog** [comment]
Here's an attempt at number two, above: https://github.com/flutter/flutter/pull/109702

**Comment by xster** [comment]
2 sounds conceptually sensible. Or otherwise, it would have been odd to bundle a scope with the navigator where in the following case 
![image](https://user-images.githubusercontent.com/156888/185550602-288e279e-b94b-40f8-ba95-d59e7b1c0ef9.png)

the traversal wouldn't go through the bottom tabs

**Comment by gspencergoog** [comment]
Should be fixed by https://github.com/flutter/flutter/pull/109702

**Comment by gspencergoog** [comment]
Actually, this doesn't appear to have fixed this issue.  Reopening.

**Comment by gspencergoog** [comment]
It looks like the root of the problem is that `MaterialPage` inserts a `PageRoute`, which is a form of `ModalRoute`.  Modal routes are supposed to capture the focus and not release it (they have a `FocusScope` in them). Pages are intended to be full screen routes, so this is correct behavior for the code in the sample. The description of `ModalRoute` is "A route that blocks interaction with previous routes."
